packages/math-extras/src/pointExtras.ts

```diff
@@ -0,0 +1,419 @@
+import { Point, IPointData, ObservablePoint } from '@pixi/math';
+
+const mixins: any = {
+    /**
+     * Adds `other` to `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method add
+     * @memberof PIXI.Point#
+     * @param {IPointData} other - The point to add to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the addition.
+     */
+    /**
+     * Adds `other` to `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method add
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} other - The point to add to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the addition.
+     */
+    add<T extends IPointData>(other: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
+        }
+        outPoint.x = this.x + other.x;
+        outPoint.y = this.y + other.y;
+
+        return outPoint;
+    },
+
+    /**
+     * Subtracts `other` from `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method subtract
+     * @memberof PIXI.Point#
+     * @param {IPointData} other - The point to subtract to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the subtraction.
+     */
+    /**
+     * Subtracts `other` from `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method subtract
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} other - The point to subtract to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the subtraction.
+     */
+    subtract<T extends IPointData>(other: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
```

Comment:
In short: Typescript is smarter than us and knows that _something that extends IPointData_ it is not exactly a `Point` and gets mad at the assignation.

Long story: `outPoint` is of type `T` and `T extends IPointData`. Since we have a check for falsy, the if only enters if you forgo or manually assign undefined to the `outPoint` parameter. By that point, you don't have a reference to write the result into so we create an object. (We can't create an object of type `T` because `T` might not have a constructor.)

Where is the catch then? You can purposefully do something awful like: `point.subtract<SomethingExtendingIPointData>(anotherPoint, undefined)`. Now your function really wants to return a `SomethingExtendingIPointData` reference but you feed undefined into the `outPoint`! We cant create a `SomethingExtendingIPointData` and `Point` wouldn't fit in there! That is what typescript is saying.

As any is a hack but I couldn't find a better way to explain typescript what's going on :(

Reply:
This might be fixed by overloading:

```
subtract<T extends IPointData>(other: IPointData, outPoint: T): T
subtract(other: IPointData): Point
```

Reply:
I could swear I tried that but let me check again 🤔

Reply:
Right, that is the problem: This is the implementation of the mixin. I can't do overloads here because this is a literal object that will be merged with the `prototype`.

On the `globals.d.ts` the overload is actually made.